### DISCUSSION TITLE
Section 1 ,No contract to fuzz. I am getting this error even on seccessfully importing libraries and making targetContract

### DISCUSSION THREAD

**Discussion by 07Vaishnavi-Singh** [Cyfrin/security-and-auditing-full-course-s23#72] [Q&A]
Error : 
 ERROR forge::multi_runner: error=No contracts to fuzz.

**Comment by johnsonstephan** [comment]
Hi @07Vaishnavi-Singh, happy to help! 

Could you please share the sequence of actions leading up to the error? What command did you run that resulted in the error?

**Comment by 07Vaishnavi-Singh** [comment]
Cool 
First I ran the fuzz test and it worked successfully telling that the if(data==2) should be changed 
Then to run the invariant test I wrote a function in the test in order to find out the
  ``` 
if(hidden==7){
    shouldAlwaysBeZero = 100 ;
} 
```
Vulnerability 
and after this the testing contract looks like this 
```
// SPDX-License-Identifier: UNLICENSED
pragma solidity ^0.8.13;

import {Test} from  "../lib/forge-std/src/Test.sol";
import "../src/MyContract.sol";
import {StdInvariant} from  "../lib/forge-std/src/StdInvariant.sol";

contract MyContractTest is StdInvariant,Test,MyContract {
 MyContract public counter;

function setup() public {
   counter = new MyContract();
   targetContract(address(counter));
}


function invariant_testInvariant() public  view{
 
    assert(counter.shouldAlwaysBeZero() == 0);

}




}
```
After this I ran the forge test command which is throwing me this error of not finding the contract to fuzz on.

**Comment by johnsonstephan** [comment]
1. Just to be sure, when you ran `forge build` did everything compile without an issue?

2. If so, then use `--match` to ensure the contract and test are clear 
- Sample code: `forge test --match-contract MyContractTest --match-test invariant_testInvariant`

**Comment by Parth-Sharma-tonchi** [comment]
1. Make sure your MyContract.sol has contract name MyContract instead of Counter.
2. And also **Try named import of MyContract in MyContractTest instead of unnamed Import**.
`Import {MyContract} from "../src/MyContract.sol";`.

3. check your test contract this is  `**setUp**` not a `setup` function. foundry didn't recognize it.
I hope it works.

**Comment by cromewar** [comment]
@07Vaishnavi-Singh does the provided solutions by @Parth-Sharma-tonchi and @johnsonstephan worked for you?
